packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -178,6 +183,72 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
       });
   }
 
+  getEntries(
+    transaction: PersistenceTransaction,
+    documentKeys: DocumentKeySet
+  ): PersistencePromise<NullableMaybeDocumentMap> {
+    return this.getSizedEntries(transaction, documentKeys).next(
+      result => result.maybeDocuments
+    );
+  }
+
+  /**
+   * Looks up several entries in the cache.
+   *
+   * @param documentKeys The set of keys entries to look up.
+   * @return A map of MaybeDocuments indexed by key (if a document cannot be
+   *     found, the key will be mapped to null) and a map of sizes indexed by
+   *     key (zero if the key cannot be found).
+   */
+  getSizedEntries(
```

Comment:
Another approach would be for `getSizedEntries` to return a `SortedMap<DocumentKey, DocumentSizeEntry>`. I decided in favor of returning two maps because it makes it easier to avoid code duplication between `getEntries` and `getSizedEntries`.

Reply:
Done, please take a look.